src/main.rs

```diff
@@ -1,101 +1,49 @@
 #![deny(unused_imports)]
+#![deny(unused_variables)]
 
+extern crate semver;
+
+mod changelog_generator;
+mod git_tag_parser;
+
+use changelog_generator::ChangelogGenerator;
+use git_tag_parser::GitTagParser;
 use std::env;
-use std::process::Command;
 use std::process::exit;
 
-struct Changelog {
-    repository_path: String,
-    from_revision: String,
-    to_revision: String
-}
+fn main() {
+    let args: Vec<_> = env::args().collect();
 
-impl Changelog {
-    pub fn generate_changelog(self) {
-        let output = self.get_log_output();
-        let lines = Changelog::get_lines_from(&output);
-        let mut lines_iterator = lines.iter();
+    if args.len() == 2 {
+        let ref repository_path = args[1];
+        let git_tag_parser = GitTagParser {
+            repository_path: repository_path.clone()
+        };
 
-        println!("## {}", self.to_revision);
-        print!("[Full Changelog](https://github.com/{}/compare/{}...{})\n\n",
-               self.get_repo_slug(), self.from_revision, self.to_revision);
+        let version_tag_pairs = git_tag_parser.get_version_tag_pairs();
 
-        loop {
-            match lines_iterator.next() {
-                Some(line) => {
-                    if line.is_empty() { return; }
-                    println!("* {}", line)
-                },
-                None => break
-            }
+        for (from_tag, to_tag) in version_tag_pairs.into_iter() {
+            let generator = ChangelogGenerator {
+                repository_path: repository_path.clone(),
+                from_revision: from_tag,
+                to_revision: to_tag
+            };
+            generator.generate_changelog();
+            print!("\n");
```

Comment:
Needed because we want each version to be separate, like this:

```markdown
## 0.0.4
[Full Changelog](https://github.com/chaos4ever/chaos/compare/0.0.3...0.0.4)

* 66b4faa chaos 0.0.4 coming up.
* 294159b Removed autochaos
* 02dc77c Various linter fixes.

## 0.0.3
[Full Changelog](https://github.com/chaos4ever/chaos/compare/0.0.2...0.0.3)

* a85b8af More refinements
* 6a1a9a7 Updated paragraph after screenshot
* 23ed01a Fixed screenshot link
```

But, you have a point - this part could be moved to the `generate_changelog` method. I'll try to fix it like that, thanks.